tests/scrape.py

```diff
@@ -401,32 +401,43 @@ def follow_button(self, button):
             raise ScrapeError('button %r has no forward URL' % button)
         return self.go(location_match.group(1))
 
-    def submit(self, region, paramdict=None, url=None, redirects=10, **params):
-        """Submit a form, optionally by clicking a given button.  The 'region'
+    def submit(self, elem, paramdict=None, url=None, redirects=10, **params):
+        """Submit a form, optionally by clicking a given button.  The 'elem'
         argument can be the form itself or a button in the form to click.
         Obtain the parameters to submit by (a) starting with the 'paramdict'
         dictionary if specified, or the default parameter values as returned
-        by get_params; then (b) adding or replacing parameters in this
+        by get_form_params; then (b) adding or replacing parameters in this
         dictionary according to the keyword arguments.  The 'url' argument
         overrides the form's action attribute and submits the form elsewhere.
         After submission, follow redirections up to 'redirects' times."""
-        form = region.tagname == 'form' and region or region.enclosing('form')
-        if not form:
-            raise ScrapeError('%r is not contained in a form' % region)
+
+        if isinstance(elem, Region):  # for backward compatibility.
+            form = elem.tagname == 'form' and elem or elem.enclosing('form')
+            if not form:
+                raise ScrapeError('%r is not contained in a form' % elem)
+            form_params = form.params
+        else:
+            try:
+                form = elem.tag == 'form' and elem or (
+                    elem.iterancestors('form')[0])
+            except IndexError:
+                raise ScrapeError('%r is not contained in a form' % elem)
+            form_params = get_form_params(form)
+
         if paramdict is not None:
             p = paramdict.copy()
         else:
-            p = form.params
-        if 'name' in region:
-            p[region['name']] = region.get('value', '')
+            p = form_params
+        if elem.get('name'):
```

Comment:
Can you add a comment that this is to add name/value of the submit button? We were first both confused what this is...